silhouette/test/com/mohiva/play/silhouette/api/services/AuthenticatorResultSpec.scala

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2015 Mohiva Organisation (license at mohiva dot com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mohiva.play.silhouette.api.services
+
+import play.api.mvc.Results
+import play.api.test.PlaySpecification
+
+/**
+ * Test case for the [[com.mohiva.play.silhouette.api.services.AuthenticatorResult]] class.
+ */
+class AuthenticatorResultSpec extends PlaySpecification {
+
+  "The `copy` method" should {
+    "return new a new instance of an authenticator result" in {
+      val result = Results.Ok
+      val authenticatorResult = AuthenticatorResult(result)
+
+      authenticatorResult.copy(result.header, result.body, result.connection) must beAnInstanceOf[AuthenticatorResult]
+    }
```

Comment:
maybe add a spec or 2 to make sure that withSession, withCookie & withHeader return AuthenticatorResult instances?